src/mame/drivers/jpmsru.cpp

```diff
@@ -287,6 +307,78 @@ void jpmsru_state::outputs_lan(address_map &map)
 	map(0x80, 0x9f).w(FUNC(jpmsru_state::out_lamp_ext_w));
 }
 
+void jpmsru_state::outputs_ews(address_map &map)
+{
+	jpmsru_io(map);
+
+	map(0x38, 0x39).w(FUNC(jpmsru_state::out_meter_w<0>));
+	map(0x3a, 0x3b).w(FUNC(jpmsru_state::out_meter_w<1>));
+	map(0x3c, 0x3d).w(FUNC(jpmsru_state::out_meter_w<2>));
+	map(0x3e, 0x3f).w(FUNC(jpmsru_state::out_meter_w<3>));
+	map(0x40, 0x41).w(FUNC(jpmsru_state::out_meter_w<4>));
+	map(0x42, 0x4d).w(FUNC(jpmsru_state::out_disp_w));
+	map(0x4e, 0x4f).w(FUNC(jpmsru_state::out_meter_w<5>));
+	map(0x50, 0x51).w(FUNC(jpmsru_state::out_meter_w<6>));
+	map(0x54, 0x55).w(FUNC(jpmsru_state::out_50p_lockout_w));
+	map(0x6a, 0x6b).w(FUNC(jpmsru_state::out_payout_cash_w));
+	map(0x6c, 0x6d).w(FUNC(jpmsru_state::out_payout_token_w));
+	map(0x6e, 0x6f).w(FUNC(jpmsru_state::out_coin_lockout_w));
+	// Mini Logic Extension outputs, used for extra lamps
+	map(0x80, 0x9f).w(FUNC(jpmsru_state::out_lamp_ext_w));
+}
```

Comment:
There’s too much repetition/duplication here.  Now you’re stuck comparing functions to work out when games have the same I/O map.

Reply:
Got rid of some of the redundant maps.